src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -777,11 +779,13 @@ private bool has_ctor_method()
             return false;
         }
 
+        [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
```

Comment:
Maybe add a comment that this is a "safe" assumption because the parent type is also annotated the same way (since the annotation talks not just about the type created here, but really about all base classes as well)